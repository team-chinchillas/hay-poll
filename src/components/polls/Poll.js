import React from 'react';
import PropTypes from 'prop-types';

export default function Poll({ poll }) {
  const inputsLists = poll.input.map(input => {
    return <li key={input}>{input}</li>;
    
  });
  const { question } = poll;
  return (
    <>
    <h2>{question}</h2>
    <ul> {inputsLists} </ul>
    </>
  );
}

Poll.propTypes = {
  poll: PropTypes.object.isRequired
};
