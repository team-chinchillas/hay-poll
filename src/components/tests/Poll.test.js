import React from 'react';
import renderer from 'react-test-renderer';
import Poll from '../polls/Poll';

describe('Poll snapshot', () => {
  it('can match a snapshot', () => {
    const poll = 
      { question: 'a question', input: ['answer1', 'answer2', 'answer3'] }
    ;
    const tree = renderer.create(
      <Poll poll={poll}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
