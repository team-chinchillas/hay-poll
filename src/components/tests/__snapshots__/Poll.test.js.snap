// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll snapshot can match a snapshot 1`] = `
Array [
  <h2>
    a question
  </h2>,
  <ul>
     
    <li>
      answer1
    </li>
    <li>
      answer2
    </li>
    <li>
      answer3
    </li>
     
  </ul>,
]
`;
